pkg/crc/preflight/preflight_daemon_task_check_windows.go

```diff
@@ -0,0 +1,138 @@
+package preflight
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/version"
+	"github.com/code-ready/crc/pkg/os/windows/powershell"
+)
+
+var (
+	// https://docs.microsoft.com/en-us/windows/win32/taskschd/daily-trigger-example--xml-
+	daemonTaskTemplate = `<?xml version="1.0" encoding="UTF-16"?>
+<Task version="1.3" xmlns="http://schemas.microsoft.com/windows/2004/02/mit/task">
+  <RegistrationInfo>
+    <Description>Run crc daemon as a task</Description>
+    <Version>%s</Version>
+  </RegistrationInfo>
+  <Settings>
+    <DisallowStartIfOnBatteries>false</DisallowStartIfOnBatteries>
+    <StopIfGoingOnBatteries>false</StopIfGoingOnBatteries>
+    <Hidden>true</Hidden>
+    <MultipleInstancesPolicy>IgnoreNew</MultipleInstancesPolicy>
+    <IdleSettings>
+      <Duration>PT10M</Duration>
+      <WaitTimeout>PT1H</WaitTimeout>
+      <StopOnIdleEnd>true</StopOnIdleEnd>
+      <RestartOnIdle>false</RestartOnIdle>
+    </IdleSettings>
+    <UseUnifiedSchedulingEngine>true</UseUnifiedSchedulingEngine>
+  </Settings>
+  <Triggers />
+  <Actions Context="Author">
+    <Exec>
+      <Command>powershell.exe</Command>
+      <Arguments>-WindowStyle Hidden -Command %s</Arguments>
+    </Exec>
+  </Actions>
+</Task>
+`
+)
+
+func genDaemonTaskInstallTemplate(crcVersion, daemonServiceName string) string {
+	return fmt.Sprintf(daemonTaskTemplate,
+		crcVersion,
+		daemonServiceName,
+	)
+}
+
+func checkIfDaemonTaskInstalled() error {
+	_, stderr, err := powershell.Execute("Get-ScheduledTask", "-TaskName", constants.DaemonTaskName)
+	if err != nil {
+		logging.Debugf("%s task is not installed: %v : %s", constants.DaemonTaskName, err, stderr)
+		return err
+	}
+	if err := checkIfOlderTask(); err != nil {
+		return err
+	}
+	return nil
+}
+
```

Comment:
here `checkIfDaemonTaskInstalled` can be simplified to just `return checkIfOlderTask()`? because that already returns an error if the call to `Get-ScheduledTask` returns an error

Reply:
I think it is much clean this way 
- Check if task installed using `Get-ScheduledTask -TaskName crcDaemon` (first block of Get-ScheduledTask) 
- Check if task is running using `$(Get-ScheduledTask -TaskName crcDaemon).Version` because if it is not running the it fail to fetch the version info
- Finally check the version is same what it should be